src/Microsoft.ML.AutoML/Experiment/MetricsAgents/RankingMetricsAgent.cs

```diff
@@ -10,11 +10,16 @@ internal class RankingMetricsAgent : IMetricsAgent<RankingMetrics>
     {
         private readonly MLContext _mlContext;
         private readonly RankingMetric _optimizingMetric;
+        private readonly int _dcgTruncationLevel;
 
-        public RankingMetricsAgent(MLContext mlContext, RankingMetric optimizingMetric)
+        public RankingMetricsAgent(MLContext mlContext, RankingMetric metric, int optimizationMetricTruncationLevel)
         {
             _mlContext = mlContext;
-            _optimizingMetric = optimizingMetric;
+            _optimizingMetric = metric;
+
+            // We want to make sure we always have at least 10 results. Getting extra results adds no measurable performance
+            // impact, so err on the side of more.
```

Comment:
Trying to distinguish between the optimization `@k` and reporting `@k`:
```suggestion
            // We want to make sure we always report metrics for at least 10 results (e.g. NDCG@10) to the user. 
            // Producing extra results adds no measurable performance impact, so we report at least 2x of the
            // user's requested optimization truncation level.
```